File: 1859-sorting-the-sentence/1859-sorting-the-sentence.cs
public class Solution
{
  public string SortSentence(string s)
  {
    string[] spl = s.Split(" ");
    Dictionary<int, string> map = new();

    foreach (string wordAll in spl)
    {
      int sequence = Convert.ToInt32(wordAll.Substring(wordAll.Length - 1));
      map.Add(sequence, wordAll.Substring(0, wordAll.Length - 1));
    }

    // System.Console.WriteLine(map);

    StringBuilder result = new();
    for (int i = 1; i < 10; i++)
    {
      result.Append(map.GetValueOrDefault(i));
      if (i != 9)
      {
        result.Append(" ");
      }
    }
    return result.ToString().TrimEnd();
  }
}

File: 1859-sorting-the-sentence/1859-sorting-the-sentence.java
class Solution {
    public String sortSentence(String s) {
        final String[] spl = s.split(" ");

        final List<String> sorted = Arrays.stream(spl).sorted((a, b) -> {
            final Integer orderA = Integer.parseInt(String.valueOf(a.charAt(a.length() - 1)));
            final Integer orderB = Integer.parseInt(String.valueOf(b.charAt(b.length() - 1)));

            return orderA - orderB;
        }).toList();


        final StringBuilder result = new StringBuilder();

        for (String str : sorted) {
            final String word = str.substring(0, str.length() - 1);
            result.append(word + " ");
        }

        return result.substring(0, result.toString().length() - 1);
    }
}
